docs/content/basics/ui_testing.md

```diff
@@ -1,4 +1,148 @@
+
+### Writing test on Espresso
+
+Let's get familiar with Espresso.
+<br/>Basically to start writing Ui tests we should:
+
+* Find the `View` (using `ViewMatchers`)
+* Interact with that `View` (using `ViewInteraction`)
+* Check state of `View` (using `ViewAssertion`)
+
+Let's start with the basic test.
+<br/>We have activity with one button (and no text shown), once we press it - text "Button pressed" shown.
+![alt text](../images/ui_testing/ui_testing_button.png "First state")![alt text](../images/ui_testing/ui_testing_button_clicked.png "Second state")
+1. We should specify which activity we should run.
+```kotlin
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import org.junit.Rule
+...
+    @get:Rule
+    var activityRule: ActivityScenarioRule<MainActivity>
+            = ActivityScenarioRule(MainActivity::class.java)
+```
+2. Create test and find our button (with `onView()` method and `withId` assertion by id)
+```kotlin
+    @Test
+    fun pressButtonAndCheckText() {
+        onView(withId(R.id.button))
+    }
+```
+
+!!! Disclaimer
+
+        Important addition - when you are trying to find some `View` that, for example, has same `id` like another View on that screen (that situation usually happens when you are trying to test screens with `RecyclerView`) you can add more `Matchers`
+        Simply place all `Matchers` that apply to single `View` on your screen inside of `allOf()` method
+        It will look like that
+        
+        <br/>@Test
+        <br/>fun someTest() {
+        <br/>&nbsp;&nbsp;&nbsp;onView(allOf(withId(R.id.button), withText("Click me")))
+        <br/>}
+
+3.Then we should perform click on it (with `click()` method from `ViewInteraction`)
+```kotlin
+    @Test
+    fun pressButtonAndCheckText() {
+        onView(withId(R.id.button))
+            .perform(click())
+    }
+```
+4. And finally, let's check that our text is shown (find the `TextView` and assert that it is displayed)
+```kotlin
+    @Test
+    fun pressButtonAndCheckText() {
+        ...
+        onView(withId(R.id.textview))
+            .check(matches(isDisplayed()))
+    }
+```
+Once we run it (run it as usual test, but you should connect real phone or run emulator), we will see standard test result
+![alt text](../images/ui_testing/ui_testing_results.png "First state")
+
+**Final code of our test**
+```kotlin
+@RunWith(AndroidJUnit4::class)
+class ExampleInstrumentedTest {
+
+    @get:Rule
+    var activityRule: ActivityScenarioRule<MainActivity>
+            = ActivityScenarioRule(MainActivity::class.java)
+
+    @Test
+    fun pressButtonAndCheckText() {
+        onView(withId(R.id.button))
+            .perform(click())
+
+        onView(withId(R.id.textview))
+            .check(matches(isDisplayed()))
+    }
+}
+```
+
+### What else?
```

Comment:
I would rather name it more meaningfully:
Espresso Cheat Sheet

Reply:
Moreover, there is the same for Compose, worth mentioning on its corresponding section
Compose Cheat sheet
https://developer.android.com/jetpack/compose/testing-cheatsheet